Libraries/LibC/netdb.cpp

```diff
@@ -294,15 +337,14 @@ void endservent()
     }
     fclose(services_file);
     services_file = nullptr;
-    keep_service_file_open = 0;
 }
 
 //Fill the service entry buffer with the information contained in the currently read line, returns true if successfull, false if failure occurs.
 static bool fill_getserv_buffer(char* line, ssize_t read)
 {
     //Splitting the line by tab delimiter and filling the servent buffers name, port, and protocol members.
     auto split_line = String(line, read).split('\t');
-    strncpy(__getserv_name_buffer, split_line[0].characters(), split_line[0].length());
+    strncpy(__getserv_name_buffer, split_line[0].characters(), split_line[0].length() + 1);
```

Comment:
Please do this correctly in the first commit instead of iterating.
Also, we're still not checking if it fits into `sizeof(__getserv_name_buffer)`